### PR TITLE
Added many objects to ProdOptimizer

### LINKED ISSUE #40: Create ProdOptimizer
Create ProdOptimizer. It needs:

DC Input and DC output specs
Various generic product objects
MPPT Arrays
THDI
CommunicationType

UNIQUE OBJECTS:

- OptimizerType (enum)
- OptimizerTypeItemType:
- - Buck Only
- - Buck-Boost
- StringPowerMax
- StringLengthMin
- StringLengthMax
- GeoRegion (string)
- IsUnlocked (boolean) //TRUE means is compatible with most inverters, FALSE means is only compatible with a list of inverters
- CompatibleInverters (is this an array or a string?)
- IsVocReducing (boolean) //some optimizers change the apparent Voc, allowing for longer string.
- IsSelectiveOptimization (boolean) // A thing where only some of the modules in an array need optimizers, not all of them. Not all Optimizers can to this
- AllowsStringConnections (boolean) // most optimizers allow only one module per input. Some allow short strings of modules (so long as them don't exceed input specs)

### PR DESCRIPTION
**Pull request by blizita**
Closes #40

### CODE REVIEW

**Review by jrippingale — APPROVED** [review]

File: Master-OB-OpenAPI.json [hunk 1]
```diff
@@ -15462,6 +15566,19 @@
           "description": "Meter per second squared"
         }
       }
+    },
+    "optimizerTypeItemType": {
+      "description": "List of Optimizer Topologies",
+      "enums": {
+        "Buck Only": {
```
Comment (cwhanse) [inline Master-OB-OpenAPI.json:15573]: No space in unit ID, space OK in label.

File: Master-OB-OpenAPI.json [hunk 2]
```diff
@@ -15462,6 +15566,19 @@
...
+          "label": "BuckOnly`",
+          "description": "This optimizer is Buck Only"
+        },
+        "Boost-Buck": {
```
Comment (cwhanse) [inline Master-OB-OpenAPI.json:15577]: No hyphen in unit ID, hypen OK in label.

File: Master-OB-OpenAPI.json [hunk 3]
```diff
@@ -10203,6 +10224,89 @@
...
+        ]
+      },
+      "IsVocReducing": {
+        "allOf": [
+          {
+            "$ref": "#/components/schemas/TaxonomyElementBoolean"
+          },
+          {
+            "type": "object",
+            "description": "This is an optimizer that modifies the attached module's apparent Voc value to allow for longer strings.",
```
Comment (cwhanse) [inline Master-OB-OpenAPI.json:10271]:
```suggestion
            "description": "True if the optimizer can modify the attached module's apparent Voc to allow for longer strings.",
```

File: Master-OB-OpenAPI.json [hunk 4]
```diff
@@ -10203,6 +10224,89 @@
...
         ]
+      },
+      "OptimizerType": {
+        "allOf": [
+          {
+            "$ref": "#/components/schemas/TaxonomyElementString"
+          },
+          {
+            "type": "object",
+            "description": "What Optimization Method Does this component Use",
```
Comment (cwhanse) [inline Master-OB-OpenAPI.json:10235]:
```suggestion
            "description": "The optimization method used by the optimizer.",
```

**Review by cwhanse — COMMENTED** [review]
Needs whatsnew addition

**Pull request merged**